src/weekday.rs

```diff
@@ -0,0 +1,170 @@
+/*
+ * Hifitime, part of the Nyx Space tools
+ * Copyright (C) 2022 Christopher Rabotin <christopher.rabotin@gmail.com> et al. (cf. AUTHORS.md)
+ * This Source Code Form is subject to the terms of the Apache
+ * v. 2.0. If a copy of the Apache License was not distributed with this
+ * file, You can obtain one at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Documentation: https://nyxspace.com/
+ */
+
+use crate::{Duration, ParsingErrors, Unit};
+use core::fmt;
+use core::ops::{Add, AddAssign, Sub, SubAssign};
+use core::str::FromStr;
+
+#[cfg(feature = "python")]
+use pyo3::prelude::*;
+
+#[cfg(feature = "serde")]
+use serde_derive::{Deserialize, Serialize};
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+#[repr(u8)]
+#[cfg_attr(feature = "python", pyclass)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum Weekday {
+    Monday = 0,
+    Tuesday = 1,
+    Wednesday = 2,
+    Thursday = 3,
+    Friday = 4,
+    Saturday = 5,
+    Sunday = 6,
+}
+
+impl Default for Weekday {
+    fn default() -> Self {
+        Self::Monday
+    }
+}
+
+impl Weekday {
+    /// Max: last weekday <=> `Sunday`, used only for conversion to/from u8.
+    const MAX: u8 = 7;
+    /// Trivial, but avoid magic numbers.
+    pub(crate) const DAYS_PER_WEEK: f64 = 7.0;
+}
+
+impl From<u8> for Weekday {
+    fn from(u: u8) -> Self {
+        match u.rem_euclid(Self::MAX) {
+            0 => Self::Monday,
+            1 => Self::Tuesday,
+            2 => Self::Wednesday,
+            3 => Self::Thursday,
+            4 => Self::Friday,
+            5 => Self::Saturday,
+            6 => Self::Sunday,
+            _ => Self::default(), // Defaults back to default for other values.
```

Comment:
this default is probably a better idea

Reply:
Yes! I agree! I didn't want to change it because you had it as Sunday, but I'm happy to set it to Monday (it makes more sense to me). It's also the most common approach around the world: https://en.wikipedia.org/wiki/Week#/media/File:First_Day_of_Week_World_Map.svg .